tests/actions.go

```diff
@@ -3796,3 +3796,72 @@ func StartValidatingAdmissionWebhookServerOrDie(context *apimachinery.CertContex
 func blockWriterPodName(info *TidbClusterConfig) string {
 	return fmt.Sprintf("%s-blockwriter", info.ClusterName)
 }
+
+// WaitForTidbComponentsReady will wait for tidbcluster components to be ready except those specified in componentsFilter.
+func (oa *OperatorActions) WaitForTidbComponentsReady(tc *v1alpha1.TidbCluster, componentsFilter map[v1alpha1.MemberType]struct{}, timeout, pollInterval time.Duration) error {
+	if tc == nil {
+		return fmt.Errorf("tidbcluster is nil")
+	}
+	var checkErr error
+	err := wait.PollImmediate(pollInterval, timeout, func() (bool, error) {
+		var local *v1alpha1.TidbCluster
+		var err error
+		tcID := fmt.Sprintf("%s/%s", tc.Namespace, tc.Name)
```

Comment:
this can be moved out of this `PollImmediate` loop?

Reply:
This implementation is copied from `WaitForTidbClusterReady`, and I think it's okay to keep it as is.

Reply:
@just1900 Then we need to update both the functions, no need to execute this during each poll, right?

Reply:
Will fix this, didn't notice that!